ratis-common/src/main/java/org/apache/ratis/util/ReferenceCountedObject.java

```diff
@@ -152,6 +153,10 @@ public boolean release() {
     };
   }
 
+  default <V> ReferenceCountedObject<V> map(Function<T, V> mapper) {
```

Comment:
Let's call it `apply(..)`, which is similar to
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#thenApply-java.util.function.Function-

Also, please add javadoc
```java
  /**
   * @return a {@link ReferenceCountedObject} by apply the given function to this object.
   */
  default <V> ReferenceCountedObject<V> apply(Function<T, V> function) {
```